tests/submap_test.py

```diff
@@ -50,12 +56,74 @@ def test_wgs84():
 
     fake_poses = np.array(
         [
-            [0.33, -0.33],
+            [0.33, -0.33, 0.0],
         ]
     )
     fake_submap_ids = [uuid.uuid3(uuid.NAMESPACE_URL, "pratt")]
     map._submap_to_utm[fake_submap_ids[0]] = (632315.689399, 4821086.845690)
 
     wgs84_coords = map.to_wgs84(fake_poses, fake_submap_ids)
     # GT computed using epsg.io
-    np.testing.assert_allclose(wgs84_coords, np.array([[43.531002, -79.3625012]]))
+    np.testing.assert_allclose(wgs84_coords, np.array([[43.531002, -79.3625012, 0.0]]))
+
+
+def test_anonymized_entry():
+    """Ensure that handling an entry with a null submap ID yields empty UTMs."""
+    blank_uuid = UUID("00000000-0000-0000-0000-000000000000")
+    map_ = Map()
+    mrp_xyz_zeros = np.array([
+        [0.0, 0.0, 0.0]
+    ])
+    utm_poses = map_.to_utm(mrp_xyz_zeros, [blank_uuid], strict=False)
+    assert np.allclose(utm_poses, np.array([[np.nan, np.nan, np.nan]]), equal_nan=True)
```

Comment:
nit: maybe `np.array_equal` would be better in these tests than `np.allclose`

Reply:
Done